src/gui/ui/opengl_gui.cpp

```diff
@@ -42,20 +42,17 @@ opengl_entry(World& world) {
     glGenVertexArrays(1, &VertexArrayID);
     glBindVertexArray(VertexArrayID);
 
-    // send color texture to gpu
-    terrain::TerrainColorMapping::assign_color_texture();
-
     render::QuadRenderer QR;
 
     ShaderHandler shader_handler = ShaderHandler();
 
-    world.update_all_chunks_mesh();
     Scene main_scene(window_width, window_height, shadow_map_size);
     setup(main_scene, world);
 
     do {
         controls::computeMatricesFromInputs(window.value());
-        main_scene.update(window.value());
+        glfwGetWindowSize(window.value(), &window_width, &window_height);
+        main_scene.update(window_width, window_width);
```

Comment:
You should unpack `window` right after you check it.

Each call to `value` is a branch.